src/internal/operations/backup.go

```diff
@@ -671,21 +671,23 @@ func (op *BackupOperation) createBackupModels(
 	sscw streamstore.CollectorWriter,
 	snapID string,
 	backupID model.StableID,
-	backupDetails *details.Details,
+	deets *details.Details,
 ) error {
-	ctx = clues.Add(ctx, "snapshot_id", snapID)
+	ctx = clues.Add(ctx, "snapshot_id", snapID, "backup_id", backupID)
 	// generate a new fault bus so that we can maintain clean
 	// separation between the errors we serialize and those that
 	// are generated during the serialization process.
 	errs := fault.New(true)
 
-	if backupDetails == nil {
+	if deets == nil {
 		return clues.New("no backup details to record").WithClues(ctx)
 	}
 
-	err := sscw.Collect(ctx, streamstore.DetailsCollector(backupDetails))
+	ctx = clues.Add(ctx, "details_entry_count", len(deets.Entries))
+
+	err := sscw.Collect(ctx, streamstore.DetailsCollector(deets))
 	if err != nil {
-		return clues.Wrap(err, "creating backupDetails persistence").WithClues(ctx)
+		return clues.Wrap(err, "creating details persistence").WithClues(ctx)
```

Comment:
```suggestion
		return clues.Wrap(err, "persisting details").WithClues(ctx)
```

Reply:
not quite, since this isn't actually the write action.  Just a collection producer.  But i did update the error to be more clear.